src/pyHepGrid/extras/grid_functions.py

```diff
@@ -0,0 +1,435 @@
+    parser.add_option("--use_cvmfs_lhapdf", action="store_true", default=True)
+    parser.add_option(
+        "--cvmfs_lhapdf_location", default="",
+        help="Provide a cvmfs location for LHAPDF.")
+    parser.add_option(
+        "--lhapdf_grid", help="absolute value of lhapdf location or relative to"
+        " gfaldir",
+        default="util/lhapdf.tar.gz")
+    parser.add_option(
+        "--lhapdf_local", help="name of LHAPDF folder local to the sandbox",
+        default="lhapdf")
+
+    # Rivet options
+    parser.add_option("--use_custom_rivet", action="store_true", default=False)
+    parser.add_option("--rivet_folder", default="Wjets/Rivet/Rivet.tgz",
+                      help="Provide the location of RivetAnalyses tarball.")
+
+    # Socket options
+    parser.add_option("-S", "--Sockets", help="Activate socketed run",
+                      action="store_true", default=False)
+    parser.add_option(
+        "-p", "--port", help="Port to connect the sockets to", default="8888")
+    parser.add_option("-H", "--Host", help="Host to connect the sockets to",
+                      default="gridui1.dur.scotgrid.ac.uk")
+
+    # Mark the run as production or warmup
+    parser.add_option("-P", "--Production", help="Production run",
+                      action="store_true", default=False)
+    parser.add_option("-W", "--Warmup", help="Warmup run",
+                      action="store_true", default=False)
+
+    parser.add_option("--pedantic", help="Enable various checks",
+                      action="store_true", default=False)
+
+    (options, positional) = parser.parse_args()
+
+    if not options.runcard or not options.runname:
+        parser.error("Runcard and runname must be provided")
+
+    if options.Production == options.Warmup:
+        parser.error(
+            "You need to enable one and only one of production and warmup")
+
+    # Pedantic checks
+    if options.Production:
+        if options.pedantic:
+            if int(options.threads) > 1:
+                parser.error("Can't run production on more than one core")
+            if int(options.threads) > 16:
+                parser.error("No node can run more than 16 threads at a time!")
+        if options.Sockets:
+            parser.error("Probably a bad idea to run sockets in production")
+
+    print_flush("Arguments: {0}".format(options))
+
+    # global options
+    global DEBUG_LEVEL
+    DEBUG_LEVEL = int(options.debug)
+
+    return options
+
+
+# ------------------------- SYSTEM UTILITIES -------------------------
+def set_default_environment(args, lhapdf_dir):
+    """
+    Initialise default environment, such that gfal works
+    """
+    do_shell(
+        "export PYTHONPATH=${PYTHONPATH}:${DIRAC}/Linux_x86_64_glibc-2.12/lib"
+        "/python2.6/site-packages")
+    os.environ["LFC_HOST"] = "lfc01.dur.scotgrid.ac.uk"
+    os.environ["LCG_CATALOG_TYPE"] = "lfc"
+    os.environ["LCG_GFAL_INFOSYS"] = "lcgbdii.gridpp.rl.ac.uk:2170"
```

Comment:
While we're on it: Can we remove these variables? I don't think they are relevant with gfal any longer.

Reply:
Go for it - as far as I'm aware they're not used.